### PR TITLE
[*] Made bill update run asynchronously and added supports for bill splitter.

### LINKED ISSUE #60: [Improvement] Asynchronous Interaction with the Database
# Initiations

- Now, our update-bill use case is very laggy.
- We need to find a way to make our program smoother.

# Design

- In the `UpdateBillUseCase`, we can create 2 threads: The querier thread and the observer thread.
- The querier thread is responsible for querying entries from the database.
- The observer thread joins the querier thread. If the querier thread finishes its jobs, then the observer thread updates the GUI.

### PR DESCRIPTION
**Pull request by ScottCTD**
# Tasks in this PR

- [x] Make the bill update use case runs asynchronously.
- [x] Extends the bill update use case to support the bill splitter. 

# Related Issues

- Close #60

### CODE REVIEW

File: src/main/java/billgates/database/MySQLDatabaseGateway.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ public MySQLDatabaseGateway() {
         this.columnToDatabaseColumn.put("From", "from");
         this.columnToDatabaseColumn.put("To", "to");
         this.columnToDatabaseColumn.put("Location", "location");
-        this.columnToDatabaseColumn.put("Splitter", "splitter");
+        this.columnToDatabaseColumn.put("Splitter", "split_bill_id");
```
Comment (Lei-Tin) [inline src/main/java/billgates/database/MySQLDatabaseGateway.java:36]: Good change! I didn't notice this lol

File: src/main/java/billgates/database/MySQLDatabaseGateway.java [hunk 2]
```diff
@@ -699,7 +699,7 @@ public void createUsersTable() {
 
 
             String query = """
-                    CREATE TABLE users
+                    CREATE TABLE IF NOT EXISTS users
```
Comment (Lei-Tin) [inline src/main/java/billgates/database/MySQLDatabaseGateway.java:702]: If we decide to do this, then maybe I will have to fix some of the tests as I manually tested for the "IF NOT EXISTS" part.

File: src/main/java/billgates/view/gui/ActionPanel.java [hunk 3]
```diff
@@ -244,6 +244,7 @@ public void view(UserJoinViewModel viewModel) {
             this.signInButton.setEnabled(false);
             this.signOutButton.setEnabled(true);
             this.addEntryButton.setEnabled(true);
+            this.backButton.setEnabled(true);
```
Comment (Lei-Tin) [inline src/main/java/billgates/view/gui/ActionPanel.java:247]: Do we want to enable the `backButton` when the user joined the application? 
Or do we want to enable the `backButton` after we go into a split bill? 

In my understanding, we should enable back button when we're in split bill, then re-disable after we get out of split bill.

**Review by Lei-Tin — APPROVED** [review]
Looks pretty cool! Just some small issues that I'm either unaware of, or something's not looking too right :3

But we can merge this already and fix the other problems later on!

File: src/main/java/billgates/view/gui/ActionPanel.java (hunk 3 above)
Comment (ScottCTD) [reply src/main/java/billgates/view/gui/ActionPanel.java:247]: Currently, we don't have a way for GUI to know the current bill.

We may improve this in the future. However, for now, we can simply enable it after the user login and disable it after logout.

**Pull request merged**